layouts/fastly/helix.vcl

```diff
@@ -10,6 +10,137 @@
 # governing permissions and limitations under the License.
 #
 
+
+# About URLs and Host headers:
+# This VCL will use the following conventions:
+# `req.url` - The canonical URL, will be used to determine the hash of the object
+# `req.http.X-Orig-URL` - The URL as received from the client
+# `req.http.X-Backend-URL` - The URL as should be sent to the origin (backend)
+#
+# Having `req.url` be canonical, vs `X-Backend-URL` being what's sent to origin 
+# allows us to do restarts to find content on other origins, and use origin
+# specific URLs, yet have the object stored under a single hash, meaning that 
+# a cache lookup will find the proper object on the first attempt once an 
+# object is in the cache.
+#
+# So `req.url` is what should be left after filtering query string parameters,
+# doing normalizations, etc. Then `req.http.X-Backend-URL` is a transformation of
```

Comment:
Is it possible that backend transformation could result in any normalization. and therefore cache efficiency?

Reply:
It shouldn't. All normalization should be done on `req.url` and `req.http.Host`.